src/showplanner/ImporterModal.tsx

```diff
@@ -14,19 +14,20 @@ export function ImporterModal(props: ImporterProps) {
   // Add support for closing the modal when the importer wants to reload the show plan.
   // There is a similar listener in showplanner/index.tsx to actually reload the show plan.
   useEffect(() => {
-    window.addEventListener(
-      "message",
-      (event) => {
-        if (!event.origin.includes("ury.org.uk")) {
-          return;
-        }
-        if (event.data === "reload_showplan") {
-          props.close();
-        }
-      },
-      false
-    );
-  });
+    function reloadListener(event: MessageEvent) {
+      if (!event.origin.includes("ury.org.uk")) {
+        return;
+      }
+      if (event.data === "reload_showplan") {
+        props.close();
+      }
+    }
+
+    window.addEventListener("message", reloadListener);
+    return () => {
+      window.removeEventListener("message", reloadListener);
+    };
+  }, [props]);
```

Comment:
Nope. This will mean that the effect gets re-created (and thus we re-add the listener) on every single prop change (and, since we're passing in a function (`props.close`), that may well be every render).

The following "fixes" it:

```suggestion
  }, [props.close]);
```

However, if the function being passed in is an anonymous function, it will have a different identity on every render, so we'll still recreate the listener. To properly fix this, wrap the `ImporterModal` callsite's `close` prop in `useCallback`.